DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountEditDetailsFormData.cs

```diff
@@ -62,5 +77,84 @@ protected MyAccountEditDetailsFormData(MyAccountEditDetailsFormData formData)
         public IFormFile? ProfileImageFile { get; set; }
 
         public bool IsDelegateUser { get; set; }
+
+        [RunValidateMethod(nameof(ValidateProfessionalRegistrationNumber))]
+        public string? ProfessionalRegistrationNumber { get; set; }
+
+        [RunValidateMethod(nameof(ValidateHasProfessionalRegistrationNumber))]
+        public YesNoSelectionEnum HasProfessionalRegistrationNumber { get; set; }
+
+        public IEnumerable<ValidationResult> ValidateHasProfessionalRegistrationNumber()
+        {
+            var validationResults = new List<ValidationResult>();
+
+            if (IsDelegateUser &&
+                HasProfessionalRegistrationNumber == YesNoSelectionEnum.None)
+            {
+                validationResults.Add(
+                    new ValidationResult(
+                        "Select your professional registration number status",
+                        new[] { nameof(HasProfessionalRegistrationNumber) }
+                    )
+                );
+
+                return validationResults;
+            }
+
+            return validationResults;
+        }
+
+        public IEnumerable<ValidationResult> ValidateProfessionalRegistrationNumber()

```

Comment:
I originally had these in seperate attributes but there is a requirement to only run this validation for delegates.
This is something we can do within the attribute but it felt less transparent.
Given we have common need, now thinking about `HEEDLS-687`, it makes more sense to have them as attributes.

Reply:
So in the interest of reusabiliy and without over complicating matters this has been moved to a helper (rather than an attribute) and is called where ever required. In this case it's called in the controller.
Registration model and my account models are different. The logic here is dependant on other factors (other model property values). I could of gone for an interface or pushed the concerning properties into a class of their own.